src/messaging/dotnet/src/Client/Client/MessageRouterClient.cs

```diff
@@ -797,11 +799,21 @@ async Task CloseTopics()
         }
     }
 
-    private ValueTask TryUnsubscribe(Topic topic)
+    private async ValueTask TryUnsubscribe(Topic topic)
     {
-        return topic.CanUnsubscribe
-            ? SendMessageAsync(new UnsubscribeMessage {Topic = topic.Name}, CancellationToken.None)
-            : default;
+        var requestId = GenerateRequestId();
+
+        try
+        {
+            if (topic.CanUnsubscribe)
+            {
+                await SendRequestAsync(new UnsubscribeMessage { RequestId = requestId, Topic = topic.Name }, CancellationToken.None);
+            }
+        }
+        catch (MessageRouterException exception)
+        {
+            _logger.LogWarning(exception, $"Exception thrown while unsubscribing, topic: {topic.Name}, request id: {requestId}.");
```

Comment:
Should we be making of use ? in here to guard against nulls in a catch logging statement?

Reply:
topic, requestId and exception are not null here, I do not see the point to guard here. but if you insist I can add. :)

Reply:
topic is never checked for null.  The first  access to it within the try can throw and then the log here will also fail if null